service/service.go

```diff
@@ -0,0 +1,143 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microendpoint/service/version"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8srestconfig"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/giantswarm/ingress-exporter/flag"
+	"github.com/giantswarm/ingress-exporter/service/collector"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+
+	Description string
+	Flag        *flag.Flag
+	GitCommit   string
+	ProjectName string
+	Source      string
+	Viper       *viper.Viper
+}
+
+type Service struct {
+	Version *version.Service
+
+	bootOnce          sync.Once
+	exporterCollector *collector.Set
+}
+
+func New(config Config) (*Service, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	// Settings.
+	if config.Flag == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Flag must not be empty", config)
+	}
+	if config.Viper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Viper must not be empty", config)
+	}
+
+	var err error
+
+	var kubernetesClient kubernetes.Interface
+	{
+		var config *rest.Config
+		config, err = rest.InClusterConfig()
```

Comment:
I would like to do this like in all other operators. This can for instance not work out locally.

Reply:
And how we do it in all other operators? Sorry my knowledge is minimal there.

Reply:
We try to keep our projects consistent. When you look in `service/service.go` in any operator, you will find what you are looking for.

Reply:
fixed